The upload method - NONE is not working

Hello,

@multiplemonomials  I found an issue with the upload method - NONE

```
[main] Configuring project: MbedCE_Testerino 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DMBED_TARGET:STRING=Wio_H725AE -DUPLOAD_METHOD:STRING=STM32CUBE -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug "-DCMAKE_C_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-gcc.exe" "-DCMAKE_CXX_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-g++.exe" -SC:/Users/USER/MbedCE/MbedCE_Testerino -Bc:/Users/USER/MbedCE/MbedCE_Testerino/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Mbed: Loading default upload method configuration from targets/upload_method_cfg/Wio_H725AE.cmake
[cmake] -- Mbed: Detected VS Code IDE, will generate VS Code debug configurations
[cmake] -- Mbed: Not building any Mbed OS tests.
[cmake] -- Located STM32CubeIDE: C:/ST/STM32CubeIDE_1.10.1/STM32CubeIDE
[cmake] -- Mbed: Code upload and debugging enabled via upload method STM32CUBE
[cmake] -- Configuring done
[cmake] -- Generating done
[cmake] -- Build files have been written to: C:/Users/USER/MbedCE/MbedCE_Testerino/build
[variant] Loaded new set of variants
[main] Configuring project: MbedCE_Testerino 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DMBED_TARGET:STRING=Wio_H725AE -DUPLOAD_METHOD:STRING=NONE -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug "-DCMAKE_C_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-gcc.exe" "-DCMAKE_CXX_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-g++.exe" -SC:/Users/USER/MbedCE/MbedCE_Testerino -Bc:/Users/USER/MbedCE/MbedCE_Testerino/build -G Ninja
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Mbed: Loading default upload method configuration from targets/upload_method_cfg/Wio_H725AE.cmake
[cmake] -- Mbed: Detected VS Code IDE, will generate VS Code debug configurations
[cmake] -- Mbed: Not building any Mbed OS tests.
[cmake] CMake Error at mbed-os/tools/cmake/mbed_ide_debug_cfg_generator.cmake:176 (list):
[cmake]   list GET given empty list
[cmake] Call Stack (most recent call first):
[cmake]   mbed-os/tools/cmake/mbed_target_functions.cmake:193 (mbed_finalize_ide_debug_configurations)
[cmake]   CMakeLists.txt:20 (mbed_finalize_build)
[cmake] 
[cmake] 
[cmake] -- Configuring incomplete, errors occurred!
[cmake] See also "C:/Users/USER/MbedCE/MbedCE_Testerino/build/CMakeFiles/CMakeOutput.log".
[cmake] See also "C:/Users/USER/MbedCE/MbedCE_Testerino/build/CMakeFiles/CMakeError.log".
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --no-warn-unused-cli -DMBED_TARGET:STRING=Wio_H725AE -DUPLOAD_METHOD:STRING=NONE -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug "-DCMAKE_C_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-gcc.exe" "-DCMAKE_CXX_COMPILER:FILEPATH=C:\Program Files (x86)\Arm GNU Toolchain arm-none-eabi\11.3 rel1\bin\arm-none-eabi-g++.exe" -SC:/Users/USER/MbedCE/MbedCE_Testerino -Bc:/Users/USER/MbedCE/MbedCE_Testerino/build -G Ninja exited with code: 1
```

I believe it is related to changes from https://github.com/mbed-ce/mbed-os/pull/129 and I think it is just about an If, but I am blind probably, sorry.

BR, Jan

--- a/tools/cmake/mbed_ide_debug_cfg_generator.cmake
+++ b/tools/cmake/mbed_ide_debug_cfg_generator.cmake
@@ -6,7 +6,7 @@
 # Detect IDEs
 # -------------------------------------------------------------
 
-if($ENV{CLION_IDE})
+if($ENV{CLION_IDE} AND MBED_UPLOAD_SUPPORTS_DEBUG)
 	message(STATUS "Mbed: Detected CLion IDE, will generate CLion debug configurations")
 	set(MBED_GENERATE_CLION_DEBUG_CFGS TRUE)
 
@@ -16,7 +16,7 @@ if($ENV{CLION_IDE})
 		message(FATAL_ERROR "In order to generate CLion configuration files, Mbed CE needs to know the name of the current CLion build profile.  This name is the string typed into the Name textbox under Settings > Build, Execution, Deployment > CMake.  Pass this name with '-DMBED_CLION_PROFILE_NAME=<name>'.")
 	endif()
 
-elseif(CMAKE_EXPORT_COMPILE_COMMANDS) # TODO: Is this actually a reliable way of detecting VS Code? Not sure if it will create false positives.
+elseif(CMAKE_EXPORT_COMPILE_COMMANDS AND MBED_UPLOAD_SUPPORTS_DEBUG) # TODO: Is this actually a reliable way of detecting VS Code? Not sure if it will create false positives.
 	message(STATUS "Mbed: Detected VS Code IDE, will generate VS Code debug configurations")
 	set(MBED_GENERATE_VS_CODE_DEBUG_CFGS TRUE)
 

--- a/tools/cmake/mbed_ide_debug_cfg_generator.cmake
+++ b/tools/cmake/mbed_ide_debug_cfg_generator.cmake
@@ -6,7 +6,7 @@
 # Detect IDEs
 # -------------------------------------------------------------
 
-if($ENV{CLION_IDE} AND MBED_UPLOAD_SUPPORTS_DEBUG)
+if("$ENV{CLION_IDE}" AND MBED_UPLOAD_SUPPORTS_DEBUG)
 	message(STATUS "Mbed: Detected CLion IDE, will generate CLion debug configurations")
 	set(MBED_GENERATE_CLION_DEBUG_CFGS TRUE)
 

